### PR TITLE
[GPU] Update strided_slice for partially dynamic shape case

### PR DESCRIPTION
**Pull request by kelvinchoi-intel**
### Details:
 - *Update strided_slice for partially dynamic shape case*
 - *Update func test to test mixed input with constant and non-constant*

### Tickets:
 - *118757*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/strided_slice.cpp [hunk 1]
```diff
@@ -45,7 +45,7 @@ std::vector<layout> strided_slice_inst::calc_output_layouts(strided_slice_node c
         || (end_data.empty() && !constant_mem.count(2))
         || (strides_data.empty() && !constant_mem.count(3))) {
         auto out_shape = ov::PartialShape::dynamic(input0_layout.get_partial_shape().size());
-        return { layout{out_shape, input0_layout.data_type, format::get_default_format(out_shape.rank().get_length())} };
+        return { layout{input0_layout.get_partial_shape(), input0_layout.data_type, format::get_default_format(out_shape.rank().get_length())} };
```
Comment (wilson-seok) [inline src/plugins/intel_gpu/src/graph/strided_slice.cpp:48]: Is there any potential issue when input rank is not same with output rank?

Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/src/graph/strided_slice.cpp]: I think there is no issue during compiling of dynamic shape. I've just used input0 layout as a default.

File: src/plugins/intel_gpu/src/graph/strided_slice.cpp [hunk 2]
```diff
@@ -44,8 +44,20 @@ std::vector<layout> strided_slice_inst::calc_output_layouts(strided_slice_node c
     if ((begin_data.empty() && !constant_mem.count(1))
         || (end_data.empty() && !constant_mem.count(2))
         || (strides_data.empty() && !constant_mem.count(3))) {
-        auto out_shape = ov::PartialShape::dynamic(input0_layout.get_partial_shape().size());
-        return { layout{out_shape, input0_layout.data_type, format::get_default_format(out_shape.rank().get_length())} };
+        auto input0_pshape = input0_layout.get_partial_shape();
+        auto input0_len = input0_pshape.size();
+        auto out_shape = ov::PartialShape::dynamic(input0_len);
+        if (input0_layout.is_dynamic()) {
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/strided_slice.cpp:50]: I think its always dynamic, isnt it?

Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/src/graph/strided_slice.cpp:50]: input0 could be static while input1/input2/input3 is dynamic

Comment (ahnyoung-paul) [reply src/plugins/intel_gpu/src/graph/strided_slice.cpp:50]: If input0 could be static, how to set out_shape? in this code, out_shape[] is not filled when input0_layout is static.
it only fills out_shape[] when input0_layout is dynamic.

Comment (yeonbok) [reply src/plugins/intel_gpu/src/graph/strided_slice.cpp:50]: Yeah at first I misread it as output. BTW I also would like to ask how static input could be handled as Paul mentioned? I think input's dynamism does not matter.

**Review by yeonbok — APPROVED** [review]

**Pull request merged**